dotCMS/src/main/java/com/dotmarketing/business/PermissionBitFactoryImpl.java

```diff
@@ -2318,7 +2266,49 @@ private void deletePermission(Permission p) {
 
 	}
 
-	private List<Permission> filterOnlyNonInheritablePermissions(List<Permission> permissions, String permissionableId) {
+    @WrapInTransaction
+	private void deleteInsertPermission(Permissionable permissionable, String type,
+            Permissionable newReference) throws DotDataException {
+
+        final String permissionId = permissionable.getPermissionId();
+
+        try{
+            Logger.debug(this.getClass(), "PERMDEBUG: " + Thread.currentThread().getName() + " - " + permissionId
+                    + " - started");
+
+            DotConnect dc1 = new DotConnect();
+            dc1.setSQL("SELECT inode FROM inode WHERE inode = ?");
+            dc1.addParam(permissionId);
+            List<Map<String, Object>> l = dc1.loadObjectResults();
+
+            dc1.setSQL("SELECT id FROM identifier WHERE id = ?");
+            dc1.addParam(permissionId);
+            List<Map<String, Object>> l2 = dc1.loadObjectResults();
+
+            if((l != null && l.size()>0) || (l2!=null && l2.size()>0)){
+                dc1.executeUpdate(DELETE_PERMISSIONABLE_REFERENCE_SQL, permissionId);
+
+                dc1.executeUpdate(INSERT_PERMISSION_REFERENCE_SQL, permissionId, newReference.getPermissionId(), type);
+            }
+
+        } catch(Exception exception){
+            if(permissionable != null && newReference != null){
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this expression which always evaluates to "true" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2589)

Reply:
do not pay attention to this

Reply:
ok